From c9ce89f6fd5fbd07cdd30a64c044afd2dc2e781b Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Thu, 18 Nov 2010 19:48:40 +0100
Subject: ASoC: ABE HAL: Add API for wakeup from retention

Reconfigure registers which may lose context when going to
retention.

Change-Id: Iaa705d0a1a29428486ed9dab248c23574356df75
Signed-off-by: sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---
 sound/soc/omap/abe/abe_api.c |   17 +++++++++++++++++
 sound/soc/omap/abe/abe_api.h |    6 ++++++
 2 files changed, 23 insertions(+), 0 deletions(-)

Index: Kernel/sound/soc/omap/abe/abe_api.c
===================================================================
--- Kernel.orig/sound/soc/omap/abe/abe_api.c	2011-01-19 00:14:18.522072429 +0530
+++ Kernel/sound/soc/omap/abe/abe_api.c	2011-01-19 00:14:35.342073087 +0530
@@ -1884,3 +1884,20 @@
 	return 0;
 }
 EXPORT_SYMBOL(abe_enable_test_pattern);
+/**
+ * abe_wakeup - Wakeup ABE
+ *
+ * Wakeup ABE in case of retention
+ */
+abehal_status abe_wakeup(void)
+{
+	/* Restart event generator */
+	abe_write_event_generator(EVENT_TIMER);
+
+	/* reconfigure DMA Req and MCU Irq visibility */
+	abe_hw_configuration();
+
+	return 0;
+}
+EXPORT_SYMBOL(abe_wakeup);
+
Index: Kernel/sound/soc/omap/abe/abe_api.h
===================================================================
--- Kernel.orig/sound/soc/omap/abe/abe_api.h	2011-01-19 00:14:18.522072429 +0530
+++ Kernel/sound/soc/omap/abe/abe_api.h	2011-01-19 00:14:35.342073087 +0530
@@ -498,5 +498,11 @@
  *
  */
 u32 abe_check_activity(void);
+/**
+ * abe_wakeup - Wakeup ABE
+ *
+ * Wakeup ABE in case of retention
+ */
+abehal_status abe_wakeup(void);
 
 #endif/* _ABE_API_H_ */
